### PR TITLE
Feature/character card UI

### PR DESCRIPTION
**Pull request by JelteDirks**
## Describe your changes

Added a character inventory on the main page where you can scroll through your characters. The scrolling uses infinite scrolling that was already present in the project, and takes up 50% of the view-height to prevent cluttering the view.

The information displayed for the character:
- key (name)
- eilodon
- level
- break effect
- crit rate
- crit dmg
- def
- effect hit rate
- effect res
- energy regen
- heal boost
- hp 
- speed

The content is displayed inside an accordion. Basic typografy from MUI is used in displaying the stats.

No further styling is applied.

## Issue or discord link
- Resolves https://github.com/frzyc/genshin-optimizer/issues/1550

## Testing/validation

Scherm­afbeelding 2024-04-17 om 13 10 46

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [x] I have made corresponding changes to README or wiki.
- [x] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

Please note that the linter for mini-ci does not support the typescript version that the project uses.

### CODE REVIEW

File: libs/sr/ui/src/Character/CharacterCard.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { CardThemed } from '@genshin-optimizer/common/ui'
+import { ICachedSroCharacter } from '@genshin-optimizer/sr/db'
+import { CardContent, Stack } from '@mui/material'
+
+export function CharacterCard({
+  character,
+}: {
+  character: ICachedSroCharacter
+}) {
+  if (!character) {
```
Comment (frzyc) [inline libs/sr/ui/src/Character/CharacterCard.tsx:10]: Proper typing should prevent this from happening, there is no need to write defensive code here against null `character`.

**Review by frzyc — CHANGES_REQUESTED** [review]
Looks like there are some lint and format issues. You can see the output from the actions and see what you can address.

File: libs/sr/ui/src/Character/CharacterCard.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}: {
+  character: ICachedSroCharacter
+}) {
+  return (
+    <Stack>
+      <CardThemed>
+        <CardContent>
+          <div>key: {character.key}</div>
+          <div>eidolon: {character.eidolon}</div>
+          <div>level: {character.level}</div>
```
Comment (nguyentvan7) [inline libs/sr/ui/src/Character/CharacterCard.tsx:16]: Need to add stat calcs here. I will come back with more guidance shortly (maybe 1d)

File: apps/sr-frontend/src/app/App.tsx [hunk 3]
```diff
@@ -30,6 +31,7 @@ export default function App() {
                 <Character />
                 <RelicEditor />
                 <Optimize />
+                <CharacterInventory />
```
Comment (nguyentvan7) [inline apps/sr-frontend/src/app/App.tsx:34]: Put this in an Accordion please

**Comment by frzyc** [comment]
If you want to link your PR to the issue, edit your description to:
```
Issue or discord link
- Resolves #1550
```

File: apps/sr-frontend/src/app/App.tsx (hunk 3 above)
Comment (nguyentvan7) [reply apps/sr-frontend/src/app/App.tsx:34]: See https://github.com/frzyc/genshin-optimizer/pull/1881/commits/3a3a79cfa8aecc6d8a03fffe8ef070ab99b731a3

Comment (nguyentvan7) [inline apps/sr-frontend/src/app/App.tsx:34]: Also move this to below `<Character />`

File: libs/sr/ui/src/Character/CharacterCard.tsx (hunk 2 above)
Comment (nguyentvan7) [reply libs/sr/ui/src/Character/CharacterCard.tsx:16]: Add these lines before the return
```
  const calc = srCalculatorWithEntries(
    withMember('member0', ...charData(character))
  )
  const member0 = convert(selfTag, { member: 'member0', et: 'self' })
```

then you can do
`calc.compute(member0.final.<stat>).val` such as `calc.compute(member0.final.atk).val`

Can probably show all of these to start, with appropriate labels

image

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]

File: libs/sr/ui/src/Character/CharacterCard.tsx (hunk 2 above)
Comment (JelteDirks) [reply libs/sr/ui/src/Character/CharacterCard.tsx]: Done

File: apps/sr-frontend/src/app/App.tsx (hunk 3 above)
Comment (JelteDirks) [reply apps/sr-frontend/src/app/App.tsx]: Done

Comment (JelteDirks) [reply apps/sr-frontend/src/app/App.tsx]: Done.

**Comment by JelteDirks** [comment]
Finished the updates request. The names of the stats might be off since I don't play the game, so please fix at your leisure.

File: libs/sr/ui/src/Character/CharacterCard.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  return (
+    <Stack>
+      <CardThemed>
+        <CardContent>
+          <Typography>Key: {character.key} </Typography>
+          <Typography>Eidolon: {character.eidolon} </Typography>
+          <Typography>Level: {character.level} </Typography>
+
+          <Typography>
```
Comment (nguyentvan7) [inline libs/sr/ui/src/Character/CharacterCard.tsx:31]: Can you add one more for `member0.final.atk` please?

**Review by nguyentvan7 — CHANGES_REQUESTED** [review]
One minor change then lgtm

**Comment by nguyentvan7** [comment]
@JelteDirks looks good, yes, you will need to merge. Your CharacterInventory will need to go in PageHome.tsx instead of App.tsx after the merge

File: libs/sr/dm/StarRailData
Comment (nguyentvan7) [inline libs/sr/dm/StarRailData:1]: This should not be changed. Should be able to run `git submodule update libs/sr/dm/StarRailData` and that should fix it

**Comment by JelteDirks** [comment]
Merged origin master into this feature. I still have the accordion since it was request, but it is on PageHome now.

File: libs/sr/dm/StarRailData
Comment (JelteDirks) [reply libs/sr/dm/StarRailData:1]: I updated the submodule (I think).

Comment (nguyentvan7) [inline libs/sr/dm/StarRailData:1]: This is still getting downgraded. 2a5b is an old commit of the submodule. Your local submodule might just be in a bad state if running the previous command didn't help. Can try running `git submodule update --remote libs/sr/dm/StarRailData`. Or we can just merge this and fix it later, it's not super critical

File: apps/sr-frontend/src/app/App.tsx [hunk 5]
```diff
@@ -53,7 +53,7 @@ function Content() {
     >
       <Header anchor="back-to-top-anchor" />
 
-      <Container maxWidth="xl" sx={{ px: { xs: 0.5, sm: 1 } }}>
+      <Container maxWidth="xl" sx={{ px: { xs: -1.5, sm: 1 } }}>
```
Comment (nguyentvan7) [inline apps/sr-frontend/src/app/App.tsx:56]: Not sure we need this. We can also fix this later

**Review by nguyentvan7 — DISMISSED** [review]

File: libs/sr/dm/StarRailData
Comment (JelteDirks) [reply libs/sr/dm/StarRailData:1]: it updated the hash now

File: apps/sr-frontend/src/app/App.tsx (hunk 5 above)
Comment (JelteDirks) [reply apps/sr-frontend/src/app/App.tsx:56]: Seems like this was done with the merge.

Comment (nguyentvan7) [reply apps/sr-frontend/src/app/App.tsx:56]: Can you undo it? Seems like a bad merge somehow, should just keep it as 0.5 padding

Comment (JelteDirks) [reply apps/sr-frontend/src/app/App.tsx]: done

File: apps/sr-frontend/src/app/PageHome.tsx [hunk 6]
```diff
@@ -10,6 +22,18 @@ export default function PageHome() {
...
+        <Accordion>
+          <AccordionSummary expandIcon={<ExpandMore />}>
+            Characters
+          </AccordionSummary>
+          <AccordionDetails>
+            <CharacterInventory />
+          </AccordionDetails>
+        </Accordion>
+      </Container>
+      <RelicEditor />
```
Comment (frzyc) [inline apps/sr-frontend/src/app/PageHome.tsx:35]: The `RelicEditor` and `RelicInventory` has been moved out of this page. can you remove it in your PR as well?

**Review by frzyc — CHANGES_REQUESTED** [review]

**Review by frzyc — APPROVED** [review]
LGTM

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**